x/datapool/keeper/pool_distribution.go

```diff
@@ -0,0 +1,239 @@
+		salesHistory.PoolId,
+		salesHistory.Round,
+		salesHistory.SellerAddress,
+	)
+	store := ctx.KVStore(k.storeKey)
+	store.Set(key, k.cdc.MustMarshalBinaryLengthPrefixed(salesHistory))
+}
+
+// GetSalesHistory returns the sales history. If there is no value, it responds nil.
+func (k Keeper) GetSalesHistory(ctx sdk.Context, poolID, round uint64, sellerAddress string) *types.SalesHistory {
+	key := types.GetKeyPrefixSalesHistory(poolID, round, sellerAddress)
+	store := ctx.KVStore(k.storeKey)
+	if !store.Has(key) {
+		return nil
+	}
+	bz := store.Get(key)
+	var salesHistory types.SalesHistory
+	k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, &salesHistory)
+	return &salesHistory
+}
+
+func (k Keeper) GetSalesHistories(ctx sdk.Context, poolID, round uint64) []*types.SalesHistory {
+	key := types.GetKeyPrefixSalesHistories(poolID, round)
+	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, key)
+	var histories []*types.SalesHistory
+	for ; iter.Valid(); iter.Next() {
+		history := &types.SalesHistory{}
+		bz := iter.Value()
+		k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, history)
+		histories = append(histories, history)
+	}
+	return histories
+}
+
+func (k Keeper) GetAllSalesHistories(ctx sdk.Context) []*types.SalesHistory {
+	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, types.KeyPrefixSalesHistory)
+	var histories []*types.SalesHistory
+	for ; iter.Valid(); iter.Next() {
+		history := &types.SalesHistory{}
+		bz := iter.Value()
+		k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, history)
+		histories = append(histories, history)
+	}
+	return histories
+}
+
+// SetInstantRevenueDistribution stores the poolID to which the revenue should be distributed immediately.
+func (k Keeper) SetInstantRevenueDistribution(ctx sdk.Context, instantRevenueDistribution *types.InstantRevenueDistribution) {
+	bz := k.cdc.MustMarshalBinaryLengthPrefixed(instantRevenueDistribution)
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.KeyPrefixInstantRevenueDistribution, bz)
+}
+
+func (k Keeper) GetInstantRevenueDistribution(ctx sdk.Context) *types.InstantRevenueDistribution {
+	store := ctx.KVStore(k.storeKey)
+	if !store.Has(types.KeyPrefixInstantRevenueDistribution) {
+		return &types.InstantRevenueDistribution{}
+	}
+	bz := store.Get(types.KeyPrefixInstantRevenueDistribution)
+	var delayedRevenueDistribution types.InstantRevenueDistribution
+	k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, &delayedRevenueDistribution)
+	return &delayedRevenueDistribution
+}
+
+// addInstantRevenueDistribution adds the poolID to distribution. If there are duplicate poolIDs, they are not added.
+func (k Keeper) addInstantRevenueDistribution(ctx sdk.Context, poolID uint64) {
+	instantRevenueDistribution := k.GetInstantRevenueDistribution(ctx)
+	instantRevenueDistribution.AppendPoolID(poolID)
+	k.SetInstantRevenueDistribution(ctx, instantRevenueDistribution)
+}
+
+func (k Keeper) DistributionRevenuePools(ctx sdk.Context) error {
```

Comment:
We use verb for name of funcs.
So, I think that verb is proper for this func as well, `DistributeRevenuePools()` or `DistributePoolsRevenue()`.